peg/kotlin_post.h

```diff
@@ -0,0 +1,161 @@
+/*
+ *   Copyright (c) 2021 Jan Dolinár
+ *
+ *   This source code is released for free distribution under the terms of the
+ *   GNU General Public License version 2 or (at your option) any later version.
+ *
+ *   This module contains functions to generate tags for Kotlin.
+ */
+
+/*
+*   INCLUDE FILES
+*/
+#include "debug.h"
+#include "options.h"
+#include "parse.h"
+#include "routines.h"
+
+#include <signal.h>
+
+/*
+* FUNCTION DEFINITIONS
+*/
+static int getcFromKotlinFile(struct parserCtx *auxil)
+{
+    int c = getcFromInputFile();
+    if (auxil->parenthesis_level > 0 && (c == '\r' || c == '\n'))
+    {
+        return ' ';
+    }
+    return c;
+}
+
+
+static void pushKind (struct parserCtx *auxil, int kind)
+{
+    intArrayAdd (auxil->kind_stack, kind);
+}
+
+static void popScope(struct parserCtx *auxil)
+{
+    tagEntryInfo *e = getEntryInCorkQueue (auxil->scope_cork_index);
+    if (e)
+        auxil->scope_cork_index = e->extensionFields.scopeIndex;
+}
+
+static void popKind (struct parserCtx *auxil, bool popScopeToo)
+{
+    intArrayRemoveLast (auxil->kind_stack);
+
+    if (popScopeToo)
+    {
+        popScope(auxil);
+    }
+}
+
+static int peekKind (struct parserCtx *auxil)
+{
+    return intArrayLast (auxil->kind_stack);
+}
+
+static void reportError (struct parserCtx *auxil)
+{
+    auxil->found_syntax_error = true;
+    fprintf(stderr, "%s: syntax error in \"%s\"\n", getLanguageName (getInputLanguage ()), getInputFileName());
+}
+
+static void makeKotlinTag (struct parserCtx *auxil, const char *name, long offset, bool pushScope)
+{
+    int k = peekKind (auxil);
+    if (k == K_IGNORE) return;
+    tagEntryInfo e;
+    if (*name != '`')
+    {
+        initTagEntry(&e, name, k);
+    } else
+    {
+        size_t len = strlen(name) - 2;
+        char *stripped = (char *)PCC_MALLOC(auxil, len+1); //TODO: Should I free it? When?
```

Comment:
Plesae, use `eMalloc(len+1)` instead of `PCC_MALLOC`.
For freeing, you can use eFree if the pointer passed to eFree is not NULL.
You can free `stripped` after calling makeTagEntry.

Reply:
Hm, interesting. I would think that the valgrind check would catch the missing `free`... Well, fixed now.